bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1639,6 +1639,12 @@ private static boolean checkObjectSubTypeForMethods(List<TypePair> unresolvedTyp
                                                         String targetTypeModule, String sourceTypeModule,
                                                         BObjectType sourceType, BObjectType targetType) {
         for (MemberFunctionType lhsFunc : targetFuncs) {
+            // As of state-2 of service typing changes, resource functions are not considered for sub-typing.
+            if (SymbolFlags.isFlagOn(targetType.getFlags(), SymbolFlags.SERVICE)
```

Comment:
Do we need to check this? Can't we rely on just the resource flag for the method?

Reply:
Yes we shold, will remove